assets/javascripts/discourse/widgets/qa-comment.js

```diff
@@ -8,24 +8,86 @@ import { ajax } from "discourse/lib/ajax";
 import bootbox from "bootbox";
 import I18n from "I18n";
 
+createWidget("qa-comment-editor", {
+  tagName: "div",
+  buildKey: (attrs) => `qa-comment-editor-${attrs.id}`,
+
+  buildClasses(attrs) {
+    return ["qa-comment-editor", `qa-comment-editor-${attrs.id}`];
+  },
+
+  defaultState(attrs) {
+    return { updatingComment: false, value: attrs.raw };
+  },
+
+  html(attrs, state) {
+    return [
+      h("textarea", attrs.raw),
+      this.attach("button", {
+        action: "editComment",
+        disabled: state.updatingComment,
+        contents: I18n.t("qa.post.edit_comment"),
+        icon: "pencil-alt",
+        className: "btn-primary qa-comment-editor-submit",
```

Comment:
Btw is this the submit or edit button? looking at the `action` and the `className`, it's different

Reply:
The way I think about it is that submit refers to the type of button while the action is the action the button takes after it has been submitted. :thinking: